### PR TITLE
Format customer currency based on the customer's country

### LINKED ISSUE #2245: Format customer currency based on the customer's country
Some currencies, like EUR, should be formatted differently based on the customer's country. Currently, we're ignoring that and using the same formatting in those cases. We should modify [load_locale_data](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/multi-currency/class-frontend-currencies.php#L149) and [get_currency_format](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/multi-currency/class-frontend-currencies.php#L133) to take the countries into account.

### PR DESCRIPTION
**Pull request by luizreis**
Fixes #2245 

#### Changes proposed in this Pull Request

* Use country-specific formatting when building currency format from locale-info
* Add get_user_locale_country method to Utils
* Use customer country in get_currency_format
* Add \_admin\_ to multi-currency register/enqueue scripts
* Update mini cart when the billing country is updated on legacy checkout

Now that we're relying on the user's locale this gets a bit simpler and all of the ~strikethrough~ removed in this edit is not relevant anymore.

#### Testing instructions

You'll need a way to change your locale to test this. You could change your site's default locale under Settings > Site Language.

*Legacy checkout*

* Enable `EUR` under the multi-currency settings
* Change the locale to `es_ES`
* Go to the checkout page and make sure that the currencies in the order summary and mini-cart are in the right format. E.g. 2.354,55€
* Navigate the Shop page and make sure the prices are in the same format
* Change the locale to `nl_NL`
* Go to the checkout page and make sure that the currencies in the order summary and mini-cart are in the right format. E.g. €2,354.55
* Navigate the Shop page and make sure the prices are in the same format

*Blocks checkout*

* Change the locale to `es_ES`
* Go to the blocks checkout and make sure that the currencies in the order summary and mini-cart are in the right format (with the exception of symbol placement). E.g. €2.354,55
* Change the locale to `nl_NL`
* Go to the blocks checkout and make sure that the currencies in the order summary and mini-cart are in the right format. E.g. €2,354.55

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/multi-currency/class-frontend-currencies.php [hunk 1]
```diff
@@ -31,9 +38,11 @@ class Frontend_Currencies {
 	 * Constructor.
 	 *
 	 * @param Multi_Currency $multi_currency The Multi_Currency instance.
+	 * @param Utils          $utils          The Utils instance.
 	 */
-	public function __construct( Multi_Currency $multi_currency ) {
+	public function __construct( Multi_Currency $multi_currency, Utils $utils ) {
```
Comment (dmallory42) [inline includes/multi-currency/class-frontend-currencies.php:43]: This is a nitpick, but what do you think about adding return type hints (where we are able to) to the functions in this class? :)

**Review by dmallory42 — APPROVED** [review]
Looks great and works well, thanks for working on this! I left one comment but it's a nitpick so feel free to ignore if you want to. 😄 

I know there were some design related questions but I'll pre-approve this from a technical point of view! 

To give my 2 💰 on the considerations you mentioned:

> The get_customer_country method initially tries to get the currency from the WC customer object, defaulting to wc_get_customer_default_location, which could take geolocation into account. Is that the desired approach or should we use something else to get the customer's country, as the current locale?

I think this is perfect, and also agree that geolocation functionality will be tackled as part of #2257.

> I opted to add get_customer_country to the Utils class, but I'd be happy to change it to something like a new Location_Service class if you find it to be better.

I think this is fine for now. If we find we're adding a lot of location related functions when doing the geolocation work, we can just move it out to a new service.

> I added a new multi-currency-checkout.js script to handle the mini-cart fragment refresh when the country is updated on the legacy checkout. Is there a better or simpler way to do this?

I'm not sure on this one, but I think the solution you ended up with is nice and simple FWIW!

File: includes/multi-currency/class-frontend-currencies.php (hunk 1 above)
Comment (luizreis) [reply includes/multi-currency/class-frontend-currencies.php:43]: Sure! I've added them in https://github.com/Automattic/woocommerce-payments/pull/2312/commits/4216424bdabeeb6a3d6f8cce71ec0f5462b1ca9e, and also fixed some wrong types in the comments.

**Comment by luizreis** [comment]
@dmallory42 - we're going to rely on the user's locale instead of the detected country to decide which formatting to present (more context in p1624640148472800/1624571480.431800-slack-CGGCLBN58). This simplifies some considerations and issues with the previous approach. I've updated the PR description and testing instructions. Would you mind taking another look at this?

File: includes/multi-currency/class-frontend-currencies.php (hunk 1 above)
Comment (dmallory42) [reply includes/multi-currency/class-frontend-currencies.php:43]: Awesome, thanks for making that change!

**Comment by dmallory42** [comment]
I had a bit of time this morning before our NIRL project kicked off so I took another look at this, it works great with the new changes - thanks for working on this @luizreis! :shipit:

**Review by dmallory42 — APPROVED** [review]

**Pull request merged**